server/server_test.go

```diff
@@ -2107,3 +2107,14 @@ func TestServerAuthBlockAndSysAccounts(t *testing.T) {
 	_, err = nats.Connect(s.ClientURL())
 	require_Error(t, err, nats.ErrAuthorization, errors.New("nats: Authorization Violation"))
 }
+
+// https://github.com/nats-io/nats-server/issues/4789
+func TestServerRestartAfterShutdown(t *testing.T) {
+	opts := DefaultTestOptions
+	opts.Port = -1
+	s := RunServer(&opts)
+	s.Shutdown()
+	s.WaitForShutdown()
+	s.Start()
+	defer s.Shutdown()
```

Comment:
Case in point, would s.Shutdown() not panic because channel `shutdownComplete` has already been closed in previous Shutdown()? My point is that I believe that there way more state that would need to be recreated/reset on a Start() after a Shutdown() that we can think of.

Reply:
If it does not panic, it means that the second Shutdown() is actually not shutting down, likely because the code thinks it is already shutdown...

Reply:
Meaning we should expand the test to make sure the server is usable? Or are you suggesting that the second start should fail and you can not re-use servers that have been shutdown?

Reply:
The latter. I think that there is a lot of "state" that would need to be recreated/reset if we wanted to support a Start() after a Shutdown() (and therefore subsequent Shutdown() calls..). The only issue I have with that is Start() does not return an error, so that would leave the user not knowing that it is not started again. We could add a StartServer() or something like that, that would return an error (similar to what we did with NewServer vs New).